modules/node_modules/@ncigdc/components/DownloadButton.js

```diff
@@ -28,14 +31,20 @@ const DownloadButton = ({
   filters = {},
   active,
   setActive,
+  altMessage = false,
+  style = {},
   extraParams = {},
+  dispatch,
 }) => {
-  const icon = <i key="icon" className={`fa fa-download ${active ? 'fa-spinner fa-pulse' : ''}`} />;
+  const icon = active ? <Spinner /> : <DownloadIcon />;
```

Comment:
I was of course referring to this icon in this button in this pull request, not making a blanket statement on all the icons on the internet :) 

"Buttons" and "Icons" are separate ideas, just as "Button" and "Text" are separate ideas. Some icons happen to also be buttons, some icons are just icons that happen to be inside buttons. Same with text (spans).

- Visible text labels inside a button labels what the button does to both sighted people and people using screen readers

- Invisible text inside a button (which is also a [perfectly](http://webaim.org/techniques/css/invisiblecontent/) [valid](http://a11yproject.com/posts/how-to-hide-content/) [method](https://snook.ca/archives/html_and_css/hiding-content-for-accessibility) for providing accessibility)  and aria-labels both communicate what a button does to only sighted people

- Icons communicate what a button does to only visual people

A button can have any one or all of those attributes. In our case, regardless of whether you go with

```html
<button>
   <div class="fa fa-download"></div>
   <div class="visually-hidden">Download</div>
</button>
```
or
```
<button aria-label="download">
   <div class="fa fa-download"></div>
</button>
```

This particular icon serves only 1 purpose in both instances (a visual label for sighted users) and would be meaningless for people using screen readers, that is what I meant :)

Your objection to my comment about how the `aria-label` would go on the button and not the icon led me to think you had intended to put it on the icon. If you see benefit in using `aria-label`s instead of visibly hidden text, and if we both agree it should go on the button, then let's put it on the button 😄